src/services/db/GitFileSystemService.ts

```diff
@@ -479,6 +483,50 @@ export default class GitFileSystemService {
   // TODO: Delete a file
   async delete() {}
 
-  // TODO: Get the latest commit of branch
-  async getLatestCommitOfBranch() {}
+  isGitLocalDiskRawCommitData(
+    commit: unknown
+  ): commit is GitLocalDiskRawCommitData {
+    return (
+      !!commit &&
+      (commit as GitLocalDiskRawCommitData).author_name !== undefined &&
+      (commit as GitLocalDiskRawCommitData).author_email !== undefined &&
+      (commit as GitLocalDiskRawCommitData).date !== undefined &&
+      (commit as GitLocalDiskRawCommitData).message !== undefined &&
+      (commit as GitLocalDiskRawCommitData).hash !== undefined
+    )
+  }
+
+  getLatestCommitOfBranch(
+    repoName: string,
+    branchName: string
+  ): ResultAsync<GitLocalDiskCommitData, GitFileSystemError> {
+    return ResultAsync.fromPromise(
+      this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).checkout(branchName).log(),
```

Comment:
nit: better to use `path.join` instead of template literals?

Reply:
no preference, used template literal to be conssitent with rest of code